cdap-runtime-ext-dataproc/src/main/java/io/cdap/cdap/runtime/spi/provisioner/dataproc/DataprocClient.java

```diff
@@ -191,134 +120,12 @@ private static DataprocClient getDataprocClient(DataprocConf conf, boolean requi
       // If the CDAP instance is running on a GCE/GKE VM from a project that matches the provisioner project,
       // use the network of that VM.
       network = systemNetwork;
-    } else if (network == null) {
-      // Otherwise, pick a network from the configured project using the Compute API
-
-      network = findNetwork(networkHostProjectId, compute);
```

Comment:
this is the main functional change. This logic still happens in ComputeNetworkClient.init(), but not in NoOpNetworkClient(). NoOpNetworkClient is used when runtime monitoring is being used (in a GCP environment) instead of SSH monitoring (in non-GCP environment). 

If the dataproc network is not given, and the network cannot be auto-detected, we now throw an exception saying the network must be provided instead of making a GCE call to pick a network.